packages/engine/src/gltf/GLTFLoader.test.tsx

```diff
@@ -0,0 +1,757 @@
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Infinite Reality Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality Engine team.
+
+All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2023 
+Infinite Reality Engine. All Rights Reserved.
+*/
+
+import { GLTF } from '@gltf-transform/core'
+import {
+  ComponentType,
+  Entity,
+  EntityContext,
+  EntityUUID,
+  UUIDComponent,
+  createEntity,
+  generateEntityUUID,
+  getComponent,
+  hasComponent,
+  setComponent,
+  useEntityContext,
+  useOptionalComponent
+} from '@ir-engine/ecs'
+import { createEngine, destroyEngine } from '@ir-engine/ecs/src/Engine'
+import {
+  NO_PROXY,
+  ReactorRoot,
+  applyIncomingActions,
+  getMutableState,
+  getState,
+  startReactor,
+  useDidMount,
+  useMutableState
+} from '@ir-engine/hyperflux'
+import { DirectionalLightComponent, PointLightComponent, SpotLightComponent } from '@ir-engine/spatial'
+import { CameraComponent } from '@ir-engine/spatial/src/camera/components/CameraComponent'
+import { RapierWorldState } from '@ir-engine/spatial/src/physics/classes/Physics'
+import { BoneComponent } from '@ir-engine/spatial/src/renderer/components/BoneComponent'
+import { MeshComponent } from '@ir-engine/spatial/src/renderer/components/MeshComponent'
+import { SceneComponent } from '@ir-engine/spatial/src/renderer/components/SceneComponents'
+import { SkinnedMeshComponent } from '@ir-engine/spatial/src/renderer/components/SkinnedMeshComponent'
+import {
+  MaterialInstanceComponent,
+  MaterialStateComponent
+} from '@ir-engine/spatial/src/renderer/materials/MaterialComponent'
+import { EntityTreeComponent, getChildrenWithComponents } from '@ir-engine/spatial/src/transform/components/EntityTree'
+import { act, render } from '@testing-library/react'
+import assert from 'assert'
+import React, { useEffect } from 'react'
+import Sinon from 'sinon'
+import { BufferGeometry, InstancedMesh, MathUtils, MeshStandardMaterial } from 'three'
+import { AssetLoaderState } from '../assets/state/AssetLoaderState'
+import { AnimationComponent } from '../avatar/components/AnimationComponent'
+import { GLTFComponent } from './GLTFComponent'
+import { GLTFDocumentState, GLTFNode, GLTFNodeState } from './GLTFDocumentState'
+import { getNodeUUID } from './GLTFState'
+import { KHRUnlitExtensionComponent, MaterialDefinitionComponent } from './MaterialDefinitionComponent'
+import { EXTMeshGPUInstancingComponent, KHRLightsPunctualComponent, KHRPunctualLight } from './MeshExtensionComponents'
+import { afterEach, beforeEach, describe, it } from 'vitest'
+
+const CDN_URL = 'https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/main/2.0'
```

Comment:
@MichaelEstes we should avoid remote dependencies for our tests. anything we use should be copied in the engine even if just for tests. any downtown or changes to the files mean our tests will start failing